### PR TITLE
[ACS -SIP Routing] GA [Hub Generated] Review request for SipRouting to add version stable/2021-05-01

### LINKED ISSUE #22678: [ACS -SIP Routing] GA - API Review
New API Review meeting has been requested.

**Service Name**: IC3 ACS Resource Provider
**Review Created By**: Jorge Garcia Hirota
**Review Date**: 2/22/2023 9:00 AM PT
**PR**: ~~https://github.com/Azure/azure-rest-api-specs/pull/22672~~ https://github.com/Azure/azure-rest-api-specs/pull/22747
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: ~~https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/dev-communication-SipRouting-2023-03-01...main/~~ https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/dzeliar-communication-SipRouting-2021-05-01...main/

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=8c46dcd2-29ae-ed11-aad1-000d3a3418aa)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

_Edit: Updated link to a new PR and description as the original branch was by mistake created as protected_

### PR DESCRIPTION
**Pull request by dzeliar**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/dzeliar-communication-SipRouting-2021-05-01...main/).
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

**Comment by dzeliar** [comment]
Hello @markweitzel, @mikekistler . I updated the version to 2023-03-01 as suggested. The preview of 2023-03-02 will be done in a separate PR as suggested. Please have a look and approve the PR. Thank you

File: specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json
Comment (mikekistler) [inline specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json:238]: You should specify `security` either globally or on each operation to associate the "azure_auth" security definition with each operation.
```suggestion
  },
  "security": [
    {
      "azure_auth": [ "user_impersonation" ]
    }
  ],
```

File: specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json
Comment (mikekistler) [inline specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json:58]: The convention in Azure is to use "update" rather than "patch" in the operationId, which is converted to a method name in generated clients. Please consider
```suggestion
        "operationId": "SipRouting_Update",
```

**Review by mikekistler — APPROVED** [review]
Looks good. 👍 

I left some suggestions but I trust your judgement on whether to take these or not.

File: specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json
Comment (dzeliar) [reply specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json:244]: Thank you @mikekistler , I added the suggested security property. We as well added HMAC authentication that's supported but was omitted in the definition

**Comment by mikekistler** [comment]
@Azure/api-stewardship-board Could we get one more review/approve on this PR?

File: specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json
Comment (weidongxu-microsoft) [reply specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json:58]: There seems no PUT. So it is CreateOrUpdate?

**Review by weidongxu-microsoft — APPROVED** [review]
I probably not in the meeting, hence not having full context.

File: specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json
Comment (dzeliar) [reply specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json:58]: Hi @weidongxu-microsoft , by default, every resource has an empty SIP configuration (we never return 404, but an empty configuration like {trunks:{}, routes:[]}). It's effectively Update vs Patch and as mentioned in my comment, on the previous reviews with the board we agreed on Patch.

**Comment by dzeliar** [comment]
Can anyone with the authority to perform merge complete this PR please?

**Comment by weidongxu-microsoft** [comment]
Let us see if @mikekistler or anyone else has more feedback.

If no, I think we can merge it tomorrow.

File: specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json
Comment (mikekistler) [reply specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json:58]: @dzeliar Normally we defer to conventions used in the service, but it appears that the convention in the Communications service is to use "update" (or in the case of one sub-service "upset") and NOT "patch".  Here's what I found in a survey of the REST APIs in the Communications service:

- operationId for patch on /chat/threads/{chatThreadId}/messages/{chatMessageId} is "ChatThread_UpdateChatMessage"
- operationId for patch on /chat/threads/{chatThreadId} is "ChatThread_UpdateChatThreadProperties"
- operationId for patch on /identities/{id} is "CommunicationIdentity_Update"
- operationId for patch on /mediaCompositions/{mediaCompositionId} is "MediaComposition_Update"
- operationId for patch on /operatorconnect/consents/{operatorId} is "CreateOrUpdateConsent"
- operationId for patch on /phoneNumbers/{phoneNumber}/capabilities is "PhoneNumbers_UpdateCapabilities"
- multiple patch operations in communicationservicejobrouter.json with operationIds that contain "upsert" (but not "patch")

If you use "patch" in your operationId, you will not only break with the Azure conventions but also the conventions used within your own service.

**Review by mikekistler — COMMENTED** [review]
The choice of "patch" for the operationId goes against both Azure's and the Communication service conventions.

File: specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json
Comment (weidongxu-microsoft) [reply specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json:58]: Yes, agree with Mike. For Azure convention, `Update` is the merge (else it would be called `CreateOrUpdate`, or `CreateOrReplace`).

Comment (dzeliar) [reply specification/communication/data-plane/SipRouting/stable/2023-03-01/communicationservicessiprouting.json]: Hello @mikekistler , @weidongxu-microsoft . I've updated operationId and naming of the dependent objects to Update (please check the[ latest commit](https://github.com/Azure/azure-rest-api-specs/pull/22747/commits/404b10e55294cd8e2d58d153a87f5d30b4dd4f37)) Please check again if it looks correct so we can get a final sign on this API. Thank you

**Comment by tjprescott** [comment]
@dzeilar please ping me when this turns green and I will merge it

**Comment by dzeliar** [comment]
Hello @mikekistler, @weidongxu-microsoft . Can you please confirm the PR looks good and is ready to merge? I updated the patch operation name to update as you requested. Once you confirm @tjprescott can merge this. Thank you

**Pull request merged**
